cmd/schema.go

```diff
@@ -74,17 +74,28 @@ func writeSchema(_ context.Context, dryRun bool, cfg *config.AppConfig) {
 
 	if viper.GetBool("mermaid") || viper.GetBool("mermaid-markdown") {
 		if cfg.SpiceDB.PolicyDir != "" {
-			files, err := os.ReadDir(cfg.SpiceDB.PolicyDir)
-			if err != nil {
-				logger.Fatalw("failed to read policy files from directory", "error", err)
-			}
+			var filePaths []string
 
-			filePaths := make([]string, 0, len(files))
+			err := filepath.WalkDir(cfg.SpiceDB.PolicyDir, func(path string, entry fs.DirEntry, err error) error {
+				if err != nil {
+					return err
+				}
 
-			for _, file := range files {
-				if !file.IsDir() && (strings.EqualFold(filepath.Ext(file.Name()), ".yml") || strings.EqualFold(filepath.Ext(file.Name()), ".yaml")) {
-					filePaths = append(filePaths, cfg.SpiceDB.PolicyDir+"/"+file.Name())
+				if entry.IsDir() {
+					return nil
 				}
+
+				ext := filepath.Ext(entry.Name())
+
+				if strings.EqualFold(ext, ".yml") || strings.EqualFold(ext, ".yaml") {
+					filePaths = append(filePaths, path)
+				}
+
+				return nil
+			})
+
+			if err != nil {
+				logger.Fatalw("failed to read policy files from directory", "error", err)
 			}
 
 			outputPolicyMermaid(filePaths, viper.GetBool("mermaid-markdown"))
```

Comment:
It looks like this code path ultimately leads to creating a policy anyway. Can we just use `NewPolicyFromDirectory` here rather than duplicate logic?

Reply:
We cannot, as `NewPolicyFromDirectory` returns a `Policy` which is a compiled policy, where as the mermaid document works off the `PolicyDocument` structure.

Once converted to a `Policy` we lose some helpful details for generating the mermaid document.

However we could create just make `NewPolicyDocumentFromDirectory` which contains the logic that both `NewPolicyFromDirectory` and the mermaid code could use.

Reply:
Split this out into `LoadPolicyDocumentFromDirectory` which now both can use.